### PR TITLE
[storage] fix that latest_state_checkpoint is messed up when there's concurrent commits

### PR DESCRIPTION
**Pull request by msmouse**
### Description

1. pass base version from executor
2. remove the cache

### Test Plan
unit tests

### CODE REVIEW

**Review by lightmark — APPROVED** [review]

**Comment by sitalkedia** [comment]
@msmouse - Thanks a lot for fixing this tricky issue. On top of passing the base version from the executor, should we also enforce that we don't have concurrent ongoing commits and if so, fail hard and fast in that case to make our life easier to debug such issues going forward?

**Comment by msmouse** [comment]
I'm on the fence -- it's nice if the db is dump and capable of going back / committing at the same time, although Zekun has changed Consensus so that it'll never happen again. I guess I'll put in a lock to fail on that for now.

File: execution/executor-benchmark/src/lib.rs [hunk 1]
```diff
@@ -166,6 +171,8 @@ mod tests {
             true,
         );
 
+        aptos_logger::info!("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!");
```
Comment (zekun000) [inline execution/executor-benchmark/src/lib.rs:174]: what is this lol

File: storage/aptosdb/src/lib.rs [hunk 2]
```diff
@@ -1164,7 +1165,8 @@ impl DbReader for AptosDB {
 
     fn get_latest_state_checkpoint(&self) -> Result<Option<(Version, HashValue)>> {
         gauged_api("get_latest_state_checkpoint_version", || {
-            Ok(self.state_store.latest_checkpoint())
+            self.state_store
+                .get_state_snapshot_before(PRE_GENESIS_VERSION - 1)
```
Comment (zekun000) [inline storage/aptosdb/src/lib.rs:1169]: this is confusing, so PRE_GENESIS_VERSION - 1  == latest?

File: storage/aptosdb/src/lib.rs [hunk 3]
```diff
@@ -1263,11 +1266,17 @@ impl DbWriter for AptosDB {
                     vec![jmt_update_refs(&jmt_updates)],
                     node_hashes.map(|hashes| vec![hashes]),
                     version,
+                    base_version,
                     &mut cs,
                 )?
-                .first()
+                .pop()
```
Comment (zekun000) [inline storage/aptosdb/src/lib.rs:1272]: what's the difference?

File: execution/executor-benchmark/src/lib.rs (hunk 1 above)
Comment (msmouse) [reply execution/executor-benchmark/src/lib.rs:174]: :trollface:

File: storage/aptosdb/src/lib.rs (hunk 2 above)
Comment (msmouse) [reply storage/aptosdb/src/lib.rs:1169]: yes

we are removing PRE_GENESIS shortly

**Pull request closed without merging**